docs/book/v3/features/container/factories.md

```diff
@@ -250,9 +250,25 @@ As an example:
 - **Requires**: no additional services are required.
 - **Optional**: no optional services are used.
 
-By default, this uses laminas-diactoros to produce a request, and will raise an
-exception if that package is not installed. You can provide an alternate factory
-if you want to use an alternate PSR-7 implementation.
+By default, this uses laminas-diactoros to produce a request, and will raise an exception if that package is not installed.
+You can provide an alternate factory if you want to use an alternate PSR-7 implementation.
+
+Also by default, this factory will consume the service `Laminas\Diactoros\ServerRequestFilter\FilterServerRequestInterface`, if defined.
+
+If you want to limit the proxy servers you trust (or supply one or more subnets), or which `X-Forwarded-*` headers are trusted, you should define the `Laminas\Diactoros\ServerRequestFilter\FilterServerRequestInterface` service; one option is to use the shipped [FilterUsingXForwardedHeadersFactory described below](#filterusingxforwardedheadersfactory).
+
+Alternately, if you want to disable usage of proxy headers entirely, alias the `FilterServerRequestInterface` to the `Laminas\Diactoros\ServerRequestFilter\DoNotFilter` class:
+
+```php
+'dependencies' => [
+    'aliases' => [
+        Laminas\Diactoros\ServerRequestFilter\FilterServerRequestInterface::class => 
+            Laminas\Diactoros\ServerRequestFilter\DoNotFilter::class,
```

Comment:
Careful: the `DoNotFilter` is not a declared service, so an alias does not suffice.

Reply:
Done.